### PR TITLE
[Lexical] Add label to run e2e tests on approve

### PR DESCRIPTION
**Pull request by Sahejkm**
- Add label and run extended test on approve

- followup on https://github.com/facebook/lexical/pull/6080

- Fixes https://github.com/facebook/lexical/issues/6097

### CODE REVIEW

**Review by potatowagon — APPROVED** [review]

**Review by etrepum — COMMENTED** [review]
LGTM, only concern is that we might want another tag for which this would not happen (e.g. an existing no-extended-tests label). We don't have anything like that yet, so I don't think it should be a blocker.

File: .github/workflows/add-label-run-extended-tests-approve.yml [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+      pull-requests: write
+    steps:
+      - run: gh pr edit "$NUMBER" --add-label "extended-tests"
+        env:
+          GH_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+          GH_REPO: ${{ github.repository }}
+          NUMBER: ${{ github.event.pull_request.number }}
+  e2e-tests:
+    needs: label_pull_requests
+    uses: ./.github/workflows/call-e2e-all-tests.yml
```
Comment (necolas) [inline .github/workflows/add-label-run-extended-tests-approve.yml:24]: I don't think this is right. Once the label is added, it should kick off the other `tests-extended` workflow automatically, as that already gets triggered when labels are added.

File: .github/workflows/add-label-run-extended-tests-approve.yml [hunk 2]
```diff
@@ -0,0 +1,24 @@
+name: Add Label to run extended tests on approve
+
+on:
+  pull_request_review:
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true
```
Comment (necolas) [inline .github/workflows/add-label-run-extended-tests-approve.yml:8]: Why cancel in progress?

File: .github/workflows/add-label-run-extended-tests-approve.yml [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+
+jobs:
+  label_pull_requests:
```
Comment (necolas) [inline .github/workflows/add-label-run-extended-tests-approve.yml:11]: The naming format used for all other workflows is `foo-bar`. So this one should probably be called something like `label-after-approved`

File: .github/workflows/add-label-run-extended-tests-approve.yml [hunk 4]
```diff
@@ -0,0 +1,24 @@
+name: Add Label to run extended tests on approve
```
Comment (necolas) [inline .github/workflows/add-label-run-extended-tests-approve.yml:1]: The file name of this workflow should be 'add-label-after-approved'. This workflow shouldn't be running e2e tests, just adding a label

**Comment by necolas** [comment]
I think the workflow file merged in this PR needs revisiting

**Comment by etrepum** [comment]
It's too bad that GitHub doesn't really offer any way to simulate these things, even to see what a matrix would expand to

File: .github/workflows/add-label-run-extended-tests-approve.yml (hunk 1 above)
Comment (Sahejkm) [reply .github/workflows/add-label-run-extended-tests-approve.yml:24]: I tested this and found out that 1 GitHub workflow action can't trigger another workflow action and hence this was needed, with a "needs" check , the second job won't run unless 1 job is not skipped. Without this tests don't trigger on approve

File: .github/workflows/add-label-run-extended-tests-approve.yml (hunk 3 above)
Comment (Sahejkm) [reply .github/workflows/add-label-run-extended-tests-approve.yml:11]: Thanks for the feedback, will update the naming conventions

**Pull request merged**